node/src/accountant/payment_adjuster/preparatory_analyser/mod.rs

```diff
@@ -495,171 +495,212 @@ mod tests {
+                .map(|weighed_account| weighed_account.analyzed_account)
+                .collect()
+        };
+        let prepare_expected_error =
             |_, total_amount_demanded_in_accounts_in_place, cw_service_fee_balance_minor| {
                 ServiceFeeImmoderateInsufficiency {
                     total_service_fee_required_minor: total_amount_demanded_in_accounts_in_place,
                     cw_service_fee_balance_minor,
                 }
             };
 
-        test_not_enough_for_even_the_least_demanding_account_causes_error(
+        test_not_enough_even_for_the_smallest_account_error(
             error_factory,
-            expected_error_preparer,
+            ensure_accounts_right_type,
+            prepare_expected_error,
         )
     }
 
     #[test]
-    fn not_enough_for_even_the_least_demanding_account_error_right_after_tx_fee_accounts_dump() {
+    fn not_enough_for_even_the_smallest_account_error_right_after_accounts_dumped_for_tx_fee() {
         let original_accounts = vec![
-            make_weighed_account(123),
-            make_weighed_account(456),
-            make_weighed_account(789),
-            make_weighed_account(1011),
+            make_meaningless_weighed_account(123),
+            make_meaningless_weighed_account(456),
+            make_meaningless_weighed_account(789),
+            make_meaningless_weighed_account(1011),
         ];
         let original_number_of_accounts = original_accounts.len();
-        let initial_sum = sum_as(&original_accounts, |account| account.balance_minor());
+        let initial_sum = sum_as(&original_accounts, |account| {
+            account.initial_balance_minor()
+        });
         let error_factory = LateServiceFeeSingleTxErrorFactory::new(&original_accounts);
-        let expected_error_preparer = |number_of_accounts, _, cw_service_fee_balance_minor| {
-            PaymentAdjusterError::LateNotEnoughFeeForSingleTransaction {
+        let ensure_accounts_right_type = |accounts| accounts;
+        let prepare_expected_error = |number_of_accounts, _, cw_service_fee_balance_minor| {
+            PaymentAdjusterError::AbsolutelyInsufficientServiceFeeBalancePostTxFeeAdjustment {
                 original_number_of_accounts,
                 number_of_accounts,
-                original_service_fee_required_total_minor: initial_sum,
+                original_total_service_fee_required_minor: initial_sum,
                 cw_service_fee_balance_minor,
             }
         };
 
-        test_not_enough_for_even_the_least_demanding_account_causes_error(
+        test_not_enough_even_for_the_smallest_account_error(
             error_factory,
-            expected_error_preparer,
+            ensure_accounts_right_type,
+            prepare_expected_error,
         )
     }
 
     #[test]
-    fn accounts_analyzing_works_even_for_weighted_payable() {
+    fn recheck_if_service_fee_adjustment_is_needed_works_nicely_for_weighted_payables() {
```

Comment:
You discovered that, `weighted` is being used here, instead of `weighed`. You might prefer to replace it in the code base using search and replace.